provider/adapters/src/fork/mod.rs

```diff
@@ -4,7 +4,179 @@
 
 //! Providers that combine multiple other providers.
 
-pub mod by_key;
+mod by_error;
+
+pub mod predicates;
 
 #[macro_use]
 mod macros;
+
+pub use by_error::ForkByErrorProvider;
+pub use by_error::MultiForkByErrorProvider;
+
+use predicates::ForkByErrorPredicate;
+use predicates::ForkByKeyPredicate;
+
+/// A provider that returns data from one of two child providers based on the key.
+///
+/// The result of the first provider that supports a particular [`DataKey`] will be returned,
+/// even if the request failed for other reasons (such as an unsupported language). Therefore,
+/// you should add child providers that support disjoint sets of keys.
+///
+/// Note: A forking [`DataProvider`] does not make sense, since there is only one key that
+/// type can support. Instead, you can create a forking [`AnyProvider`], [`BufferProvider`],
```

Comment:
typo

Reply:
I don't see the typo

Reply:
Actually not typo, but still style. "that" can be parsed as either the determiner of the relative clause "that [the] type can support", in which case this clause is missing an article, or as a demonstrative article of the noun phrase if you're dropping the determiner ("[that] that type can support"). My parser defaults to the first one and has to backtrack, making this a bumpy read.

Out of all the combinations of 

> only one key (that|which|) (that|this|the) type can support

"that" is the only one that is ambiguous.

Reply:
I re-wrote the paragraph.